api/bolt/datastore.go

```diff
@@ -85,9 +88,28 @@ func NewStore(storePath string, fileService portainer.FileService) (*Store, erro
 		store.checkForDataMigration = true
 	}
 
+	// Backup database
+	dbBackupPath := path.Join(storePath, dbBackupFileName)
```

Comment:
since this path is not used anywhere, and in 2.0 we have different backup mechanism. I'd suggest:
1. move logic to backup1_24db (so it's clear it can be removed in the future)
2. we don't need the simpleCopyFile function, just put inside the backup function
3. log errors to console `log.Printf`, and log at the end of the process something like "backup for 1.24 finished successfly"
4. check the current version before backup